magicblock-committor-service/src/stubs/changeset_committor_stub.rs

```diff
@@ -1,152 +1,178 @@
-use solana_sdk::{hash::Hash, signature::Signature};
+use solana_sdk::signature::Signature;
+use solana_transaction_status_client_types::{
+    EncodedConfirmedTransactionWithStatusMeta, EncodedTransaction,
+    EncodedTransactionWithStatusMeta,
+};
 use tokio::sync::oneshot;
 
 use crate::{
     error::CommittorServiceResult,
-    persist::{
-        BundleSignatureRow, CommitStatus, CommitStatusRow,
-        CommitStatusSignatures, CommitStrategy, CommitType,
+    intent_execution_manager::{
+        BroadcastedIntentExecutionResult, ExecutionOutputWrapper,
     },
-    ChangesetCommittor,
+    intent_executor::ExecutionOutput,
+    persist::{CommitStatusRow, IntentPersisterImpl, MessageSignatures},
+    service_ext::{BaseIntentCommitorExtResult, BaseIntentCommittorExt},
+    types::{ScheduledBaseIntentWrapper, TriggerType},
+    BaseIntentCommittor,
 };
 
 #[derive(Default)]
 pub struct ChangesetCommittorStub {
     reserved_pubkeys_for_committee: Arc<Mutex<HashMap<Pubkey, Pubkey>>>,
     #[allow(clippy::type_complexity)]
-    committed_changesets: Arc<Mutex<HashMap<u64, (Changeset, Hash, bool)>>>,
+    committed_changesets: Arc<Mutex<HashMap<u64, ScheduledBaseIntentWrapper>>>,
 }
 
-#[async_trait]
-impl ChangesetCommittor for ChangesetCommittorStub {
-    fn commit_changeset(
+impl ChangesetCommittorStub {
+    #[allow(clippy::len_without_is_empty)]
+    pub fn len(&self) -> usize {
+        self.committed_changesets.lock().unwrap().len()
+    }
+}
+
+impl BaseIntentCommittor for ChangesetCommittorStub {
+    fn reserve_pubkeys_for_committee(
         &self,
-        changeset: Changeset,
-        ephemeral_blockhash: Hash,
-        finalize: bool,
-    ) -> oneshot::Receiver<Option<String>> {
-        static REQ_ID: AtomicU64 = AtomicU64::new(0);
-        let reqid = REQ_ID.fetch_add(1, Ordering::Relaxed);
-        let (tx, rx) = tokio::sync::oneshot::channel();
-        self.committed_changesets
+        committee: Pubkey,
+        owner: Pubkey,
+    ) -> oneshot::Receiver<CommittorServiceResult<Instant>> {
+        let initiated = Instant::now();
+        let (tx, rx) = oneshot::channel::<CommittorServiceResult<Instant>>();
+        self.reserved_pubkeys_for_committee
             .lock()
             .unwrap()
-            .insert(reqid, (changeset, ephemeral_blockhash, finalize));
-        tx.send(Some(reqid.to_string())).unwrap_or_else(|_| {
-            log::error!("Failed to send commit changeset response");
+            .insert(committee, owner);
+
+        tx.send(Ok(initiated)).unwrap_or_else(|_| {
+            log::error!("Failed to send response");
         });
         rx
     }
 
+    fn commit_base_intent(
+        &self,
+        base_intents: Vec<ScheduledBaseIntentWrapper>,
+    ) {
+        let mut changesets = self.committed_changesets.lock().unwrap();
+        base_intents.into_iter().for_each(|intent| {
+            changesets.insert(intent.inner.id, intent);
+        });
+    }
+
+    fn subscribe_for_results(
```

Comment:
Not sure why we needed to switch from basic oneshot (handling result in one place) instead to returning an emitter to broadcast to multiple handlers (which seems complex), but if it works then ok.

Reply:
Could explain what is inefficient and unidiomatic here? IntentExecutionEngine accepts intents, schedules execution and streams/broadcasts the results, thats the only thing it does